tests/test_workflows.py

```diff
@@ -1567,6 +1574,19 @@ def test_change_properties_after_compilation(self):
                 maxiter=100, method="vgd", jac="finite_difference"
             )
 
+    def test_numpy_serialize(self):
+        np_qubo = NumberPartition([1, 2, 3]).qubo
+
+        q = QAOA()
+        q.compile(np_qubo)
+        q.optimize()
+
+        # add numpy results
+        numpy_res = {"000": np.int64(85), "100": np.int64(85), "010": np.int64(85)}
```

Comment:
@vishal-ph 

we could change this tests by turning the `cost` also into a numpy float

And similarly for the intermediate step. Should we?